front/app/containers/Admin/pagesAndMenu/containers/ContentBuilder/components/CraftSections/HomepageBanner/index.tsx

```diff
@@ -70,10 +106,104 @@ const HomepageBannerSettings = () => {
         node.data.props.homepageSettings.banner_avatars_enabled,
       banner_signed_in_header_multiloc:
         node.data.props.homepageSettings.banner_signed_in_header_multiloc,
+      banner_cta_signed_in_text_multiloc:
+        node.data.props.homepageSettings.banner_cta_signed_in_text_multiloc,
+      banner_cta_signed_in_type:
+        node.data.props.homepageSettings.banner_cta_signed_in_type,
+      banner_cta_signed_in_url:
+        node.data.props.homepageSettings.banner_cta_signed_in_url,
+      banner_cta_signed_out_text_multiloc:
+        node.data.props.homepageSettings.banner_cta_signed_out_text_multiloc,
+      banner_cta_signed_out_type:
+        node.data.props.homepageSettings.banner_cta_signed_out_type,
+      banner_cta_signed_out_url:
+        node.data.props.homepageSettings.banner_cta_signed_out_url,
     },
   }));
 
   const [search, setSearchParams] = useSearchParams();
+  const { formatMessage } = useIntl();
+
+  const labelMessages: Record<CTASignedOutType, MessageDescriptor> = {
+    customized_button: messages.customized_button,
+    no_button: messages.no_button,
+    sign_up_button: messages.sign_up_button,
+  };
+
+  const handleUrlChange = (
+    value: string,
+    field: 'banner_cta_signed_out_url' | 'banner_cta_signed_in_url'
+  ) => {
+    const validation = isValidUrl(value);
+    setProp((props: Props) => (props.homepageSettings[field] = value));
+
+    if (!validation) {
+      setProp((props: Props) => {
+        const newErrorTypes = props.errorTypes?.includes(field)
+          ? [...props.errorTypes]
+          : [...(props.errorTypes || []), field];
+
+        props.errorTypes = newErrorTypes;
+        props.hasError = true;
+        eventEmitter.emit(CONTENT_BUILDER_ERROR_EVENT, {
+          [id]: {
+            hasError: true,
+          },
+        });
+      });
+    } else {
+      setProp((props: Props) => {
+        const newErrorTypes = props.errorTypes?.filter(
+          (errorType) => errorType !== field
+        );
+        props.errorTypes = newErrorTypes || [];
+        if (newErrorTypes && newErrorTypes.length === 0) {
+          props.hasError = false;
+          eventEmitter.emit(CONTENT_BUILDER_ERROR_EVENT, {
+            [id]: {
+              hasError: false,
+            },
+          });
+        }
+      });
+    }
+  };
+
+  const handleCtaTypeChange = (
+    value: CTASignedOutType | CTASignedInType,
+    field: 'banner_cta_signed_out_type' | 'banner_cta_signed_in_type'
+  ) => {
+    setProp(
+      (props: Props) =>
+        (props.homepageSettings[field] = value as CTASignedInType)
```

Comment:
`value` could also be CTASignedOutType though right? Can we remove this typecasting somehow?